x-pack/plugins/alerts/server/alerts_client.ts

```diff
@@ -245,9 +243,19 @@ export class AlertsClient {
         }
         throw e;
       }
-      await this.unsecuredSavedObjectsClient.update('alert', createdAlert.id, {
-        scheduledTaskId: scheduledTask.id,
-      });
+      await this.unsecuredSavedObjectsClient.create<RawAlert>(
```

Comment:
We may also have to partially update the alert status when the alert fails to decrypt (setting error status in this case)..

Reply:
Ya, already looking at that - I think we're cool - we seem to be doing partial updates for other non-AAD things like `muted`.  Biggest change is that we need to update the SO with the kibana system user, which is something I don't think we've had to deal with yet, but this doesn't seem like a problem either.

The alert status will both not be decrypted, and added to the non-AAD field list, so we _can_ do a partial update, and the status can be retrieved even if decryption would have failed.

Reply:
I think we should encapsulate the usage of partial updates and limit it to the places we **have** to do it.
I definitely don't want any public APIs (such as the client) which are doing partial updates as it has unexpected results that client do no expect or know about.

It seems safer to avoid partial updates where we can and then be **very explicit** about it when we do.

In this case we don't **need** to do a partial update, as we already have all the data we need- so I'd suggest we keep it a full update.